### PR TITLE
Make tab label length and window title length configurable

### PR DESCRIPTION
**Pull request by 00AR**
Implements #3344 
Added a GtkSpinButton in "Notebook tabs" in Interface preferences. This will take as input, number of characters that the user wants to display on the tab label.
For now, Maximum value that can be entered: 100000
**Default value**: 99999(as suggested by @elextr to be very big number)
Minimum value that can be entered: 3
The value is stored in the variable **tab_label_len** declared in GeanyInterfacePrefs struct. This value will be written in **geany.conf** when **save_dialog_prefs** of keyfile.c is executed.
The tab label will get updated dynamically on clicking the Apply/Ok button in the Preferences dialog.
Screenshot from 2023-01-04 11-34-19

### CODE REVIEW

**Review by eht16 — CHANGES_REQUESTED** [review]
Apart from the few comments, it looks very good.

By accident I found https://github.com/geany/geany/issues/2112 which would be solved by this as well.

Maybe you could add "closes #2112 and #3344" to the commit message and rephrase it a bit as "display size" might be confused with the size of the monitor. Maybe something like "Make tab label length configurable".

File: src/document.c [hunk 1]
```diff
@@ -434,7 +434,7 @@ void document_update_tab_label(GeanyDocument *doc)
 
 	g_return_if_fail(doc != NULL);
 
-	short_name = document_get_basename_for_display(doc, -1);
+	short_name = document_get_basename_for_display(doc, interface_prefs.tab_label_len);
```
[the remark this answers is not in the document]
Comment (elextr) [reply src/document.c:437]: The window title is usually wider than an individual tab, so using the same setting is not really the best thing.  But at least using the setting would make it variable.

[the remark this answers is not in the document]
Comment (00AR) [reply src/document.c:437]: I wanted to keep the changes minimum and relevant to issue #3344 (my first PR :D) that is why I avoided touching window title. But I will make this change.

**Comment by 00AR** [comment]
I missed this, does the documentation also needs to be updated?

**Comment by elextr** [comment]
> does the documentation also needs to be updated?

Yes, just add the description in geany.txt where that dialog is described, @eht16 will add the dialog to his list of images to be updated before release ;-)

**Comment by eht16** [comment]
Thanks for the changes, @00AR.

LGTM. I will merge in a few days if there will be no objections.

**Comment by 00AR** [comment]
> LGTM. I will merge in a few days if there will be no objections.

Alright. Thanks!

**Pull request merged**